File: Data Structures & Algorithms/valid-sudoku/submission-0.py
class Solution:
    def isValidSudoku(self, board: List[List[str]]) -> bool:
        # row[i] = [()]
        # col[j] = [()]
        # block[i//3, j//3] = [()]
        # not in row or not in col or not in block
        rowLength = len(board)
        colLength = len(board[0])
        rows = [set() for _ in range(rowLength)]
        cols = [set() for _ in range(colLength)]
        numOfBlocks = (rowLength//3) * (colLength//3)
        block = [set() for _ in range(numOfBlocks)]
        
        for i in range(rowLength):
            for j in range(colLength):
                blockI, blockJ = i//3, j//3
                blockNum = (blockI * 3) + blockJ
                if board[i][j] == ".":
                    continue
                if board[i][j] in rows[i]:
                    print(rows)
                    return False
                if board[i][j] in cols[j]:
                    print(cols)
                    return False
                if board[i][j] in block[blockNum]:
                    print(block)
                    return False
                rows[i].add(board[i][j])
                cols[j].add(board[i][j])
                block[blockNum].add(board[i][j])
        return True
                    

File: Data Structures & Algorithms/search-for-word/submission-0.java
public class Solution {
    private int ROWS, COLS;
    boolean ans = false;
    
    public boolean exist(char[][] board, String word) {
        ROWS = board.length;
        COLS = board[0].length;
        

        for (int r = 0; r < ROWS; r++) {
            for (int c = 0; c < COLS; c++) {
                dfs(board, word, r, c, 0);
            }
        }
        return ans;
    }

    private void dfs(char[][] board, String word, int r, int c, int i) {
        if (i == word.length()) {
            ans = true;
            return;
        }
        if (r < 0 || c < 0 || r >= ROWS || c >= COLS || 
            board[r][c] != word.charAt(i) || board[r][c] == '#') {
            return;
        }

        board[r][c] = '#';
        dfs(board, word, r + 1, c, i + 1);
        dfs(board, word, r - 1, c, i + 1);
        dfs(board, word, r, c + 1, i + 1);
        dfs(board, word, r, c - 1, i + 1);
        System.out.println(r + " , " + c);
        board[r][c] = word.charAt(i);
    }
}